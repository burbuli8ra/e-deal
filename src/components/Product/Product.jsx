import React from 'react';
import Styled from './Product.style'

const Product = ({ product }) => {
  return (
    <Styled.Product>
      <Styled.Title>{product.title}</Styled.Title>
      <Styled.Image
        alt={product.title}
        src={product.image_url}
      />
      <Styled.Price>{product.price/100}â‚¬</Styled.Price>

      <Styled.Label>Features</Styled.Label>
      <Styled.Text
        dangerouslySetInnerHTML={{ __html: product.excerpt || '-' }}
      />

      <Styled.Label>Description</Styled.Label>
      <Styled.Text
        dangerouslySetInnerHTML={{ __html: product.description || '-' }}
      />
    </Styled.Product>
  );
};

export default Product;