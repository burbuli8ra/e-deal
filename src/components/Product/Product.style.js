import styled from '@emotion/styled';
import { colors } from 'theme';

const Label = styled.div`
  color: ${colors.pastelBlue};
  font-size: 18px;
  font-weight: 700;
  margin-bottom: 4px;
  text-decoration: underline;
`;
Label.displayName = 'Label';

const Image = styled.img`
  border-radius: 4px;
  height: auto;
  width: auto;
  margin-bottom: 30px;
  max-height: 320px;
  max-width: 320px;
`;
Image.displayName = 'Image';

const Price = styled.div`
  color: ${colors.darkPastelBlue};
  font-size: 36px;
  font-weight: 700;
  margin-top: 0;
  margin-bottom: 30px;
`;
Price.displayName = 'Price';

const Product = styled.div`
  align-items: center;
  align-self: flex-start;
  display: flex;
  flex-direction: column;
`;
Product.displayName = 'Product';

const Title = styled.h2`
  color: ${colors.darkPastelBlue};
  font-size: 24px;
  font-weight: 700;
  margin-top: 0;
  margin-bottom: 30px;
`;
Title.displayName = 'Title';

const Text = styled.div`
  color: ${colors.pastelBlue};
  margin-bottom: 30px;
  text-align: center;
  
  &:last-of-type {
    margin-bottom: 0;
  }
  
  > * {
    &:first-of-type {
      margin-top: 0;
    }
    
    &:last-of-type {
      margin-bottom: 0;
    }
  }
`;
Text.displayName = 'Text';

export default {
  Label,
  Image,
  Price,
  Product,
  Title,
  Text
};