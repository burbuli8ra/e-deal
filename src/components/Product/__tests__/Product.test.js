import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Mocks from '__mocks__';
import Product from '../Product';

const setup = props => (
  <Router>
    <Product {...props} />
  </Router>
);

describe('Product component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = { product: Mocks.Product };
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});