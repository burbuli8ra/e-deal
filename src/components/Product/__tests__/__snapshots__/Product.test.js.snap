// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1c7yiq"
  >
    <h2
      class="css-8pegrd"
    >
      Viking Apollo F-1
    </h2>
    <img
      alt="Viking Apollo F-1"
      class="css-1ssd2ae"
      src="https://abpcdn.pstatic.gr/P/bpimg54/3uyUu/1JkFGD_SX959Y938/0/2155960749.webp"
    />
    <div
      class="css-17agv4d"
    >
      470€
    </div>
    <div
      class="css-1rvkpvg"
    >
      Features
    </div>
    <div
      class="css-mb3fio"
    >
      <b>
        Viking
      </b>
      , Μέγιστο βάρος χρήστη: 120 Kg, Βάρος τροχού: 9 Kg
    </div>
    <div
      class="css-1rvkpvg"
    >
      Description
    </div>
    <div
      class="css-mb3fio"
    >
      -
    </div>
  </div>
</DocumentFragment>
`;
